cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/DefaultPatternCompiler.java

```diff
@@ -0,0 +1,14 @@
+package io.cucumber.cucumberexpressions;
+
+import java.util.regex.Pattern;
+
+/**
+ * Default {@link PatternCompiler} compiling {@link Pattern} with flag {@link Pattern#UNICODE_CHARACTER_CLASS}
+ */
+class DefaultPatternCompiler implements PatternCompiler {
+
+	@Override
+	public Pattern compile(String regexp) {
+		return Pattern.compile(regexp, Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
This method should have two arguments. The Android implementation can then filter anything it doesn't support while the the default passes everything on.

Reply:
To be exact it should have the same signature as: `Pattern compile(String regex, int flags)`

Given the difference in the JDK and Android implementation I imagine that these problems will pop up more often in the future.